### DISCUSSION TITLE
Is there a "findorcreate" or "upsert" method for EntityManagers?

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#196] [Q&A] [answered]
I'm using MongoDB, which has this type of behavior built-in, and I'm not sure if Typetta support for this.

**Comment by minox86** [comment, marked as answer]
There's no upsert method yet because it's a method specific of MongoDB and hard to implement on the SQL side. Anyway, you can use any MongoDB specific method building raw queries as described [here](https://twinlogix.github.io/typetta/chapters/client/raw-databse-access#raw-queries).
